### ISSUE TITLE
infer units of quantities

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#498]
This was started by @AKM11 (see #377 for some details, and code in `Language.Drasil.Misc`), but not quite finished. Definitely a good feature to have.  As #377 what about a different problem, I'm going to close that one, and the current issue is for remembering to finish the feature itself.

**Comment by elwazana** [comment]
@JacquesCarette Hello Dr. Carette is there any way for me to help with this issue?
 
I've looked through #377 and the `Language.Drasil.Misc` file's relevant code; if I can work on finishing this issue, I would like some clarification on what exactly needs finishing in the feature as you mentioned above as that is still unclear to me.

**Comment by JacquesCarette** [comment]
Eventually, yes. However, @halonazhao is working on things relating to units right now, and so it would be quite dangerous to do this in parallel.  The likelyhood of a difficult merge are way too high.

**Comment by balacij** [comment]
So, I'm not immediately sure why we want to infer units of quantities. Is it to check that a symbol declaration's unit matches the unit of an expression we're trying to equate it to?

**Comment by JacquesCarette** [comment]
Yes, it is all about checking coherence. Some expressions might well have units that don't make sense (the same way things can be ill-typed).
